packages/medusa/src/utils/clean-response-data.ts

```diff
@@ -0,0 +1,21 @@
+import { pick } from "lodash"
+
+/**
+ * Filter response data to contain props specified in the fields array.
+ *
+ * @param data - record or an array of records in the response
+ * @param fields - record props allowed in the response
+ */
+function cleanResponseData<T>(data: T, fields: string[]) {
+  if (!fields.length) {
+    return data
+  }
+
+  if (Array.isArray(data)) {
```

Comment:
Question : since pick is doing a deep clone of entities, does the entities contain extra functions etc? Does it can alter the performances under load?

Reply:
I guess a shallow clone here would be sufficient no?

Reply:
it's hard to say how would this impact performance without measuring it, but yes I agree that a shallow clone is enough - and I thought that `pick` is doing exactly that, cloning only top level object, no?